rust/arrow/src/memory.rs

```diff
@@ -16,12 +16,125 @@
 // under the License.
 
 //! Defines memory-related functions, such as allocate/deallocate/reallocate memory
-//! regions.
+//! regions, cache and allocation alignments.
 
 use std::alloc::Layout;
 use std::mem::align_of;
 
-pub const ALIGNMENT: usize = 64;
+// NOTE: Below code is written for spatial/temporal prefetcher optimizations. Memory allocation
+// should align well with usage pattern of cache access and block sizes on layers of storage levels from
+// registers to non-volatile memory. This alignments are all cache aware alignments incorporated
+// from [cuneiform](https://crates.io/crates/cuneiform) crate. Approach mimicks Intel TBB's
+// cache_aligned_allocator which to exploit cache locality and minimize prefetch signals with
```

Comment:
This sentence does not read right to me.  Maybe "This approach mimicks Intel TBB's cache_aligned_allocator which exploits cache locality and minimizes prefetch signals resulting in less ..."